encoding/toml.ts

```diff
@@ -549,39 +549,41 @@ function joinKeys(keys: string[]): string {
     .join(".");
 }
 
+enum ArrayType {
+  ONLY_PRIMITIVE,
+  ONLY_OBJECT_EXCLUDING_ARRAY,
+  MIXED,
+}
+
 class Dumper {
   maxPad = 0;
   srcObject: Record<string, unknown>;
   output: string[] = [];
+  _arrayTypeCache = new Map<unknown[], ArrayType>();
```

Comment:
I suggest to use private field (prefixed with `#` instead of `_`)

Reply:
👌 Do you think we should update methods starting with `_` too? Maybe for the whole project?

Reply:
I think we should use `#` for methods too.